### PR TITLE
Implement lazy tag rendering

### PR DESCRIPTION
**Pull request by gschier**
This PR implements lazy rendering of Nunjucks tags. 

_beside the benefit discussed below, this is also required for request chaining (#210)_

## Problem

When the environment contains a tag that throws an error, the request will fail to execute even if the variable is not referenced. This is because the entire environment is rendered on its own (to support recursive variables) before it's used to render the request. Here's an example:

```javascript
// environment
{
  "broken": "{% badname %}"
}
```

The request will fail with a render error even if it does not contain `{{ broken }}`.

## Solution

During the environment rendering phase, only render tags and not variables. Then, when rendering the request, render both variables and tags.

## To-Do 

- [x] Implement a way to skip rendering of tags (by telling Nunjucks a different tag syntax)
- [x] Render environments with `variablesOnly` set to `true`
- [x] Add test for `variablesOnly` mode
- [x] Render request twice in case a variable contains a tag as a value
  - `eg. {{ foo }} => {% uuid 'v4' %} => dd265685-16a3-4d76-a59c-e8264c16835a`

### CODE REVIEW

File: app/common/render.js [hunk 1]
```diff
@@ -32,68 +28,98 @@ export async function buildRenderContext (ancestors, rootEnvironment, subEnviron
...
+  // Do an Object.assign, but render each property as it overwrites. This
+  // way we can keep same-name variables from the parent context.
   const renderContext = {};
   for (const environment of environments) {
-    // Do an Object.assign, but render each property as it overwrites. This
-    // way we can keep same-name variables from the parent context.
-    await _objectDeepAssignRender(renderContext, environment);
+    // Sort the keys that may have Nunjucks last, so that other keys get
+    // defined first. Very important if env variables defined in same obj
+    // (eg. {"foo": "{{ bar }}", "bar": "Hello World!"})
```
Comment (gschier) [inline app/common/render.js]: This just inlines the _objectDeepAssignRender function

File: app/ui/components/modals/request-render-error-modal.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ class RequestRenderErrorModal extends PureComponent {
 
     const result = jq.query(request, `$.${error.path}`);
     const template = (result && result.length) ? result[0] : null;
-    const locationLabel = template.includes('\n')
+    const locationLabel = template && template.includes('\n')
```
Comment (gschier) [inline app/ui/components/modals/request-render-error-modal.js]: This was a minor unrelated bug fix.

File: app/common/render.js [hunk 3]
```diff
@@ -32,68 +28,98 @@ export async function buildRenderContext (ancestors, rootEnvironment, subEnviron
...
+  // Make a deep copy so no one gets mad :)
+  const newObj = clone(obj);
 
-  // Make a copy so no one gets mad :)
   async function next (x, path = '') {
     if (blacklistPathRegex && path.match(blacklistPathRegex)) {
       return x;
     }
 
+    const asStr = Object.prototype.toString.call(x);
```
Comment (sean3z) [inline app/common/render.js]: 👍 rather than continuously invoking `toString()`

Comment (gschier) [inline app/common/render.js]: Ya, I don't know what old-me was thinking before 😳
